projects/fsastorefrontlib/src/cms-components/user-change-address/user-change-address.component.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Address, Country, Region, UserAddressService } from '@spartacus/core';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import { OccValueListService } from '../../occ/services/value-list/occ-value-list.service';
+import { ConsentService } from '../../core/my-account/facade/consent.service';
+import { FSUser } from '../../occ/occ-models/occ.models';
+
+@Component({
+  selector: 'cx-fs-user-change-address-dialog',
+  templateUrl: './user-change-address.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class UserChangeAddressComponent implements OnInit, OnDestroy {
+  @Input() customer: FSUser;
+  @Input() userId: string;
+  @Output() actionChange = new EventEmitter<string>();
+  addressForm: FormGroup = this.fb.group({
+    country: this.fb.group({
+      isocode: [null, Validators.required],
+    }),
+    line1: ['', [Validators.required]],
+    line2: [''],
+    town: ['', [Validators.required]],
+    postalCode: ['', [Validators.required]],
+  });
+  countriesURL = '/catalogs/financialProductCatalog/valueLists/country';
+  countries$: Observable<Country[]>;
+  selectedCountry$: BehaviorSubject<string> = new BehaviorSubject<string>('');
+  regions$: Observable<Region[]>;
+  private subscription = new Subscription();
+
+  constructor(
+    protected fb: FormBuilder,
+    protected occValueListService: OccValueListService,
+    protected userAddressService: UserAddressService,
+    protected fsConsentService: ConsentService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.customer?.defaultAddress) {
+      this.addressForm.patchValue(this.customer?.defaultAddress);
+      this.selectedCountry$.next(this.addressForm.get('country.isocode').value);
+    }
+    this.countries$ = this.occValueListService
+      .getValuesFromAPI(this.countriesURL)
+      .pipe(
+        filter(result => result.values),
+        map(result => {
+          const options: Country[] = [];
+          result.values.forEach(item => {
+            options.push({
+              name: item.value,
+              isocode: item.key,
+            });
+          });
+          return options;
+        })
+      );
+    this.regions$ = this.selectedCountry$.pipe(
```

Comment:
When a country is selected, here I dynamically add/remove region form control.

Reply:
I saw their solution and it works in the same way. I thought this is more semantically and functionally correct since the region control should exist only if the country which has a region is selected, otherwise it shouldn't. With enable/disable works only because when transferring data from the form to address object `this.addressForm.value` and it won't take the disabled value (which is the case when a country without region is selected) even region control exists and contains a value. But if I write something like this: `this.addressForm.getRawValue()` then it will take the disabled value. This is a similar example to our dynamic forms where we used enable/disable for the hiding/show form field. I can use enable/disable, but I think the name says when we should use it. Here I need to show/hide form control depending on the other form element.

Reply:
I was thinking more from the perspective of less and clear code. To me at least, disabled/enabled approach is more clear when I look at the code at first glance. I completely agree that enable/disable wasn't build for that purpose, and that this is some kind of hack, but in this context I would use it. As far as I can see both approaches have the same result on the UI and DOM. I don't see use-case for us getting the raw value for any reason. You decide.